rsocket/RSocket.cpp

```diff
@@ -12,17 +12,19 @@ folly::Future<std::unique_ptr<RSocketClient>> RSocket::createConnectedClient(
     std::shared_ptr<RSocketStats> stats,
     std::shared_ptr<RSocketConnectionEvents> connectionEvents,
     std::shared_ptr<ResumeManager> resumeManager,
-    std::shared_ptr<ColdResumeHandler> coldResumeHandler) {
+    std::shared_ptr<ColdResumeHandler> coldResumeHandler,
+    folly::EventBase* stateMachineEvb) {
```

Comment:
I'd vote for having this be the first arg, not the last.

Reply:
I believe most common usage of RSocketClient will be simple, where the user lets the RSocketClient::StateMachine use the Transport's EventBase.   I suspect only in sophisticated scenarios the user would want to explicitly control the StateMachine's EventBase.